modules/shortcodes.php

```diff
@@ -118,12 +118,51 @@ function jetpack_preg_replace_callback_outside_tags( $pattern, $callback, $conte
 	
 	$textarr = wp_html_split( $content );
 	unset( $content );
+
+	// Flag to check we're inside an HTML tag.
+	$inside_tag = '';
```

Comment:
This could be just me, but when I read `$inside_tag`, my mind automatically thinks, "this must store the tag that is contained within another tag" but I believe that the intention is to specify, "this is the tag that we are currently inside."

In other contexts, I've more often seen this named `$outside_tag` or `$surrounding_tag` or something like that than `$inside_tag`.

Another choice could be to separate them out for clarity: `$is_inside` and `$parent_tag` or what-not.